data/data/vsphere/master/main.tf

```diff
@@ -1,44 +1,64 @@
 locals {
-  description = "Created By OpenShift Installer"
+  description  = "Created By OpenShift Installer"
+  vcenter_key  = keys(var.vsphere_vcenters)[0]
+  template_map = { for t in var.template : t.name => t }
 }
 
 provider "vsphere" {
-  user                 = var.vsphere_username
-  password             = var.vsphere_password
-  vsphere_server       = var.vsphere_url
+  user                 = var.vsphere_vcenters[local.vcenter_key].user
+  password             = var.vsphere_vcenters[local.vcenter_key].password
+  vsphere_server       = var.vsphere_vcenters[local.vcenter_key].server
   allow_unverified_ssl = false
 }
 
+data "vsphere_datacenter" "datacenter" {
+  count = var.master_count
+  name  = var.vsphere_control_planes[count.index].workspace.datacenter
+}
+
+data "vsphere_resource_pool" "resource_pool" {
+  count = var.master_count
+  name  = var.vsphere_control_planes[count.index].workspace.resourcePool
+}
+
+data "vsphere_datastore" "datastore" {
+  count         = var.master_count
+  name          = var.vsphere_control_planes[count.index].workspace.datastore
+  datacenter_id = data.vsphere_datacenter.datacenter[count.index].id
+}
+
 resource "vsphere_virtual_machine" "vm_master" {
   count = var.master_count
 
   name                 = "${var.cluster_id}-master-${count.index}"
-  resource_pool_id     = var.resource_pool
-  datastore_id         = var.datastore
-  num_cpus             = var.vsphere_control_plane_num_cpus
-  num_cores_per_socket = var.vsphere_control_plane_cores_per_socket
-  memory               = var.vsphere_control_plane_memory_mib
-  guest_id             = var.guest_id
-  folder               = var.folder
-  enable_disk_uuid     = "true"
-  annotation           = local.description
+  resource_pool_id     = data.vsphere_resource_pool.resource_pool[count.index].id
+  datastore_id         = data.vsphere_datastore.datastore[count.index].id
+  num_cpus             = var.vsphere_control_planes[count.index].numCPUs
+  num_cores_per_socket = var.vsphere_control_planes[count.index].numCoresPerSocket
+  memory               = var.vsphere_control_planes[count.index].memoryMiB
```

Comment:
I think these (cpus, cores, mem) are consistent across control plane nodes right? But resource pool, datastore, and folder CAN be different (when its multizone)? If that's the case, it might be more clear to not index these.

Reply:
Go back to sending as variables to terraform or just `[0]` ?

Reply:
Just moved it to [0], if you want something else let me know